Applications/Spire/Source/Ui/TabWidget.cpp

```diff
@@ -0,0 +1,71 @@
+#include "Spire/Ui/TabWidget.hpp"
+#include <QKeyEvent>
+#include <QTabBar>
+#include "Spire/Spire/Dimensions.hpp"
+
+using namespace Spire;
+
+TabWidget::TabWidget(QWidget* parent)
+    : QTabWidget(parent),
+      m_last_focus_was_key(false) {
+  tabBar()->setFixedHeight(scale_height(40));
+  setStyleSheet(QString(R"(
+    QWidget {
+      outline: none;
+    }
+
+    QTabWidget::pane {
+      border: none;
+    }
+
+    QTabBar::tab {
+      background-color: #EBEBEB;
+      font-family: Roboto;
+      font-size: %1px;
+      height: %2px;
+      margin: %3px %4px %3px 0px;
+      width: %5px;
+    }
+
+    QTabBar::tab:focus {
+      border: %6px solid #4B23A0;
+      padding: -%6px 0px 0px -%6px;
+    }
+
+    QTabBar::tab:hover {
+      background-color: #DBDBDB;
+      color: black;
+    }
+
+    QTabBar::tab:selected {
+      background-color: #F5F5F5;
+      color: #4B23A0;
+    })").arg(scale_height(12)).arg(scale_height(20)).arg(scale_height(10))
+        .arg(scale_width(2)).arg(scale_width(80)).arg(scale_width(1)));
+  connect(this, &QTabWidget::tabBarClicked, this,
+    &TabWidget::on_tab_bar_clicked);
+  connect(this, &QTabWidget::currentChanged, this,
+    &TabWidget::on_tab_changed);
+}
+
+void TabWidget::keyPressEvent(QKeyEvent* event) {
+  if(event->key() == Qt::Key_Left || event->key() == Qt::Key_Right) {
+    m_last_focus_was_key = true;
+  }
+}
+
+void TabWidget::mousePressEvent(QMouseEvent* event) {
+  m_last_focus_was_key = false;
+}
+
+void TabWidget::on_tab_bar_clicked(int index) {
+  if(widget(index) != nullptr) {
```

Comment:
if(auto selectedWidget = widget(index)) {
      selectedWidget->setFocus();
    }

Why is focus explicitly being given? Why doesn't Qt give focus automatically?

Reply:
I meant to remove these before opening the PR. The current uses of QTabWidget used this functionality but it doesn't cause a noticeable effect in the test widget.

Reply:
When I update the current QTabWidget usage to use this TabWidget I'll evaluate if they're required.